src/nvim/api/tabpage.c

```diff
@@ -122,6 +122,45 @@ Window nvim_tabpage_get_win(Tabpage tabpage, Error *err)
   abort();
 }
 
+/// Sets the current window in a tabpage
+///
+/// @param tabpage  Tabpage handle, or 0 for current tabpage
+/// @param win Window handle, must already belong to {tabpage}
+/// @param[out] err Error details, if any
+void nvim_tabpage_set_win(Tabpage tabpage, Window win, Error *err)
+  FUNC_API_SINCE(12)
+{
+  tabpage_T *tp = find_tab_by_handle(tabpage, err);
+  if (!tp) {
+    return;
+  }
+
+  win_T *wp = find_window_by_handle(win, err);
+  if (!wp) {
+    return;
+  }
+
+  tabpage_T *win_tp = win_find_tabpage(wp);
+
+  if (win_tp != tp) {
+    api_set_error(err, kErrorTypeException, "Window does not belong to tabpage %d", tp->handle);
+    return;
+  }
+
+  if (tp == curtab) {
+    try_start();
+    goto_tabpage_win(win_tp, wp);
+    if (!try_end(err) && wp != curwin) {
+      api_set_error(err,
+                    kErrorTypeException,
+                    "Failed to switch to window %d",
+                    wp->handle);
+    }
+  } else {
+    tp->tp_curwin = wp;
```

Comment:
Might want to set `tp_prevwin` to the old `tp_curwin` here too, but only if `tp->tp_curwin != wp` (to avoid `tp_prevwin == tp_curwin`).

Reply:
Thanks for the reviews! Will add this in now.